### ISSUE TITLE
Test errors in DispatcherBlockConsumerTest

### ISSUE THREAD

**Issue by merlimat** [apache/pulsar#5438]
There are several open issues for intermettent test failures in `DispatcherBlockConsumerTest`. 
#506, #786, #1082 and #814. 

After a closer look, it seems to me that in some cases, the unblocking of consumers is not working as expected (eg. after consumer sends redelivery requests) and the tests are passing because of a race condition that masks the problem.

cc/ @rdhabalia

**Comment by tisonkun** [comment]
Closed as stale and no recent reports.

Feel free to open a new issue if it's still relevant to the maintained versions.

**Comment by asafm** [comment]
@codelipenghui If this masks a failure, shouldn't we open this as issue?
